lib/exq/middlewares/manager.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Exq.Middlewares.Manager do
+  @behaviour Exq.Middlewares.Behaviour
+  require Logger
+  alias Exq.Manager.Server, as: Manager
+
+  def before_work(pipeline) do
+    pipeline
+  end
+
+  def after_processed_work(pipeline) do
+    notify(pipeline)
+    pipeline
+  end
+
+  def after_failed_work(pipeline) do
+    notify(pipeline)
+    pipeline
+  end
+
+
+  defp notify(pipeline) do
+    case is_alive?(pipeline.assigns.manager) do
```

Comment:
I think with supervisor order above we don't need the `alive?` checks so can probably remove a bunch of code from here (and methods below).